### DISCUSSION TITLE
Body Padding

### DISCUSSION THREAD

**Discussion by Ivan-Chupin** [umputun/remark42#1465] [General]
I've noticed that Remark42 doesn't fit perfectly into the frame intended for it. Remark.css -> body{padding:6px} was to blame. I think this is wrong, because such indents should be carried out exclusively by the layout designer of the site.

**Comment by akellbl4** [comment]
It's intentional because we have lots of small elements that have different decorations that can go out of that frame. If we don't have that padding you'll see clipped elements that do not fit.

You can do `.wrapper { position: relative; overflow: hidden; }; #remark42 { margin: -6px }` if you don't want that padding.

We will try to address this request in design updates.

**Reply by Ivan-Chupin** [reply]
> It's intentional because we have lots of small elements that have different decorations that can go out of that frame.

How can Remark42 go outside the iframe it's in?
![padding](https://user-images.githubusercontent.com/83099323/186500771-fc77951b-5563-4782-acf9-c50aba8a3c59.png)

I removed the body{padding:6px} in Remark.css and Reamark42 behaved as expected by the web designer. However, I did not notice absolutely no problems. An example can be found [here](https://smarthdd.com).

@akellbl4 Can you describe a scenario where you would see the need to use body{padding:6px} in remark.css?

**Reply by akellbl4** [reply]
I offer to add negative margin on wrapper of the iframe on the page and not inside the iframe

An example: <img width="207" alt="CleanShot 2022-08-24 at 12 11 36@2x" src="https://user-images.githubusercontent.com/2330682/186503269-8db81b70-89f0-4d8e-9901-d66c568ad4d9.png">

**Reply by Ivan-Chupin** [reply]
May be add?
```
.sort-picker.A_A {
    padding-right: 2px;
}
```

**Reply by akellbl4** [reply]
1. It's not only one example. I had specific reason to add 6px 
2. For this example it's not critical but for other parts element will be misaligned.

**Reply by paskal** [reply]
If that's a known improvement point, do we want to convert it from discussion to issue?

**Reply by akellbl4** [reply]
To be honest I don't see any improvments here. Also, I don't get a point how this breaks anyting.

**Reply by paskal** [reply]
@Ivan-Chupin, could you please build a case for why this should be changed? I think a jsfiddle with a demonstration of bad and desired behaviour would be ideal for that purpose.

**Reply by Ivan-Chupin** [reply]
@paskal Unfortunately, I have no experience with jsfiddle, and besides, it is not available in the country where I am located. Therefore, I will try to explain in words and pictures.

If I put a Remark42 in a `<div></div>` then I naturally expect the Remark42 elements to align to the edges of the div. But since Remark42 uses a 6 point padding from each side of the div, this breaks the layout design.

Left as it is now, right as it should be:
![Remark42-Padding-2](https://user-images.githubusercontent.com/83099323/187035542-247adf81-323a-4851-bcd7-428c4662a4b7.png)
Why does Remark42 use body{padding: 6px}? Why is the padding not 2 or 10? The answer is absolutely obvious - because it is a crutch that masks the incorrect layout of Remark42.

**Reply by paskal** [reply]
I've created https://github.com/umputun/remark42/issues/1487 about it, I think it's a valid issue.
